### PR TITLE
Fix: Fix the table population problem when the skip outliers switch is activated/deactivated

### PR DESCRIPTION
**Pull request by d-bernat**
please see https://dhis2.atlassian.net/browse/DHIS2-16375

### CODE REVIEW

File: dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/outlierdetection/AnalyticsOutlierDetectionController.java [hunk 1]
```diff
@@ -144,4 +156,11 @@ private OutlierRequest getFromQuery(OutlierQueryParams queryParams, boolean anal
 
     return request;
   }
+
+  private void checkAnalyticsTableForOutliers() {
```
Comment (maikelarabori) [inline dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/outlierdetection/AnalyticsOutlierDetectionController.java:160]: Would be nice to have a e2e to test this error. Please see the class `NoAnalyticsTablesErrorsScenariosTest`. You can add a new test there to simulate this exception and assert the response/error.

**Review by maikelarabori — CHANGES_REQUESTED** [review]
Hi @d-bernat, the fix looks good. I added two comments that would be nice to address.

Cheers

File: dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/outlierdetection/AnalyticsOutlierDetectionController.java (hunk 1 above)
Comment (d-bernat) [reply dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/outlierdetection/AnalyticsOutlierDetectionController.java:160]: e2e is coming, there are no e2e for outliers

Comment (maikelarabori) [reply dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/outlierdetection/AnalyticsOutlierDetectionController.java:160]: Okay, if you prefer to add it as part of all others e2e for Outliers is fine. Just please add a TODO in the class `NoAnalyticsTablesErrorsScenariosTest`, so we do not forget this scenario.

[the remark this answers is not in the document]
Comment (d-bernat) [reply dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/outlierdetection/AnalyticsOutlierDetectionController.java]: done

**Review by maikelarabori — APPROVED** [review]

**Comment by maikelarabori** [comment]
Hi @d-bernat, please do not forget to update the PR description with a proper one. Thanks.

**Review by gnespolino — APPROVED** [review]

File: dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/outlierdetection/AnalyticsOutlierDetectionController.java (hunk 1 above)
Comment (d-bernat) [reply dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/outlierdetection/AnalyticsOutlierDetectionController.java]: done

**Pull request merged**